db/database.go

```diff
@@ -138,6 +138,7 @@ type DatabaseContextOptions struct {
 	SessionCookieHttpOnly     bool             // Pass-through DbConfig.SessionCookieHTTPOnly
 	AllowConflicts            *bool            // False forbids creating conflicts
 	SendWWWAuthenticateHeader *bool            // False disables setting of 'WWW-Authenticate' header
+	DisablePublicBasicAuth    bool             // True enforces OIDC/guest only
```

Comment:
Have reviewed and looks good if the answers to the questions in Jira are both "No"

But will wait for clarification around guest/sessions first, as I suspect we'll need to cover those additional cases.